### PR TITLE
feedback to pulp3 migration guide

### PR DESCRIPTION
**Pull request by ryandeussing**
This PR includes all the changes from #663 and #707 , _including the modifications made after review_.

ttps://issues.redhat.com/browse/SAT-4175

Cherry-pick into:

* [x] Foreman 3.0
* [x] Foreman 2.5 (Satellite 6.10)
* [ ] Foreman 2.4
* [ ] Foreman 2.3 (Satellite 6.9)
* [ ] Foreman 2.1 (Satellite 6.8)

### CODE REVIEW

**Comment by swadeley** [comment]
Hi, the problem with making new PRs with merging the older one is there is no way people can be sure their changes are in the new PR without spending lots of time checking, so this is something to be avoided.

**Comment by ryandeussing** [comment]
This is understood. This is undesirable and and an ant-pattern and to be avoided.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 1]
```diff
@@ -17,11 +17,46 @@ Cloning is the recommended method because that prevents them being overwritten i
...
+----
+. Restart the `{foreman-maintain}` services using:
++
+[options="nowrap",  subs="verbatim,quotes,attributes"]
+----
+# systemctl daemon-reload
+# {foreman-maintain} service restart
+----
++
+For further information, see xref: migrating_pulp_content[Migrating Content to Pulp 3].
```
Comment (jlsherrill) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:54]: we need to wrap this whole section in a ` ifdef::satellite[]`

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc [hunk 2]
```diff
@@ -4,6 +4,12 @@
 
 This section describes how to upgrade {ProjectServer} from {ProjectVersionPrevious} to {ProjectVersion}.
 You can upgrade from any minor version of {ProjectName} Server {ProjectVersionPrevious}.
+However, to access the latest migration and tooling fixes, {Team} recommends that you upgrade to the latest minor version of {ProjectServer} before upgrading from {ProjectVersionPrevious} to {ProjectVersion}.
+
+Although the process of upgrading from {ProjectVersionPrevious} to {ProjectVersion} also migrates Pulp content, this can take some considerable time.
+You can reduce this time by following the procedure in xref:migrating_pulp_content[]
+
+If you have not pre-migrated content from Pulp 2 to Pulp 3, you must upgrade to the latest version of {ProjectServer} before starting to migrate Pulp content.
```
Comment (jlsherrill) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc:12]: this will also need to be wrapped in a ` ifdef::satellite[]`

**Comment by jlsherrill** [comment]
2 small comments, everything else looks good

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
+[[id="migrating_pulp_content"]]
+
+= Migrating Content to Pulp 3
+
+The time to prepare content for Pulp 3 depends on the amount of content and the number of content views.
+For large systems, this can mean several days of downtime.
+To prevent this, {Team} recommends that you pre-migrate Pulp content while running the latest version of {ProjectServer} {ProjectVersionPrevious}.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:7]: I was told `recommend` is a problematic wording.

Maybe as an alternative say something like "pre-migrate content to reduce downtime/avoid overly long update procedures"?

cc @melcorr

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+This reduces the overall upgrade downtime.
+
+Use this procedure to begin migrating content from Pulp 2 to Pulp 3.
+
+.Procedure
+
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:13]:
```suggestion
```

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+Migrated/Total RPMs: 0/367633
+Migrated/Total errata: 0/20780140
+Migrated/Total repositories: 0/33924
+Estimated migration time based on yum content: 47 hours, 23 minutes
+Note: ensure there is sufficient storage space for /var/lib/pulp/published to
+double in size before starting the migration process.
+Check the size of /var/lib/pulp/published with 'du -sh /var/lib/pulp/published/'
+[OK]
+----
+. Update the file permissions before the {ProjectServer} upgrade using the following:
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:40]:
```suggestion
. Update the file permissions before upgrading {ProjectServer} using the following:
```

I think we don't say "*the* Foreman/Satellite/orcharhino".

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
++
+This might take some time on high latency systems.
+. Prepare your content for Pulp migration using the following command:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {foreman-maintain} content-prepare
+----
++
+You cannot use `*Ctrl*` *+* `*C*` to terminate this process. If a user attempts to halt the process using CTRL + C or by disconnecting their SSH session, the process does not terminate but continues in the background.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:55]:
```suggestion
You cannot use `*Ctrl*` *+* `*C*` to terminate this process.
If a user attempts to halt the process using CTRL + C or by disconnecting their SSH session, the process does not terminate but continues in the background.
```

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+You can use the following command to terminate the process gracefully, whenever necessary, so that you can continue later.
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {foreman-maintain} content-prepare abort
+----
++
+Note that `{foreman-maintain} content-prepare abort` can take several minutes to terminate the process.
+. Continue the migration process using `{foreman-maintain} content-prepare` whenever it is convenient.
+. If problems occur you can restart the pre-migration process from the beginning:
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:65]:
```suggestion
. If problems occur, you can restart the pre-migration process from the beginning:
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc [hunk 8]
```diff
@@ -147,16 +147,10 @@ endif::[]
...
-----
-# {foreman-maintain} service restart
-----
 . If you have migrated content from Pulp 2 to Pulp 3, remove all Pulp 2 content.
 +
 [options="nowrap" subs="attributes"]
 ----
 # {foreman-maintain} content remove-pulp2
 ----
+This removes Pulp 2 RPMs, content in /var/lib/pulp/content/, the mongo database, and migration content in the Pulp 3 database.
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]:
```suggestion
This removes Pulp 2 RPMs, content in `/var/lib/pulp/content/`, the mongo database, and migration content in the Pulp 3 database.
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc [hunk 9]
```diff
@@ -147,16 +147,10 @@ endif::[]
 ----
 # {foreman-maintain} service restart
 ----
-. Optional: If you made manual edits to DNS or DHCP configuration files, check and restore any changes required to the DNS and DHCP configuration files using the backups that you make.
-. If you make changes in the previous step, restart the `{foreman-maintain}` services.
-+
-[options="nowrap" subs="attributes"]
-----
-# {foreman-maintain} service restart
-----
```
Comment (maximiliankolb) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]: This looks like an unrelated change.

**Review by maximiliankolb — COMMENTED** [review]
A couple of suggestions and questions, esp. regarding using the word"recommend".

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc (hunk 3 above)
Comment (melcorr) [reply guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:7]: You're correct @maximiliankolb 
IBM Style, which is the standard with which all existing docs were written and reviewed against, cautions against "recommend" on pg 357:

"recommend v
Avoid if possible. A phrase such as “We recommend that you take the following action” could create a
potential marketing or legal problem."
Thank you for contributions!

Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:7]: revmoved

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc (hunk 9 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]: removed

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc (hunk 2 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_parent.adoc:12]: added

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc (hunk 1 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:54]: added

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc (hunk 9 above)
Comment (maximiliankolb) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]: Am I wrong or does this PR still remove those lines, which I consider an unrelated change.

Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]: I removed them on this PR, is that what you are confirming?

Comment (maximiliankolb) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]: It's part of "upgrading Foreman"; how does this relate to the pulp3 migration?

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+# {foreman-maintain} content-prepare
+----
++
+You cannot use `*Ctrl*` *+* `*C*` to terminate this process.
+If a user attempts to halt the process using CTRL + C or by disconnecting their SSH session, the process does not terminate but continues in the background.
+You can use the following command to terminate the process gracefully, whenever necessary, so that you can continue later.
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {foreman-maintain} content-prepare abort
```
Comment (swadeley) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:60]:
```suggestion
# {foreman-maintain} content prepare-abort
```

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+----
++
+Note that `{foreman-maintain} content-prepare abort` can take several minutes to terminate the process.
```
Comment (swadeley) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:63]:
```suggestion
Note that `{foreman-maintain} content prepare-abort` can take several minutes to terminate the process.
```

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+----
+# {foreman-maintain} prep-{TargetVersion}-upgrade
+----
++
+This might take some time on high latency systems.
+. Prepare your content for Pulp migration using the following command:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {foreman-maintain} content-prepare
```
Comment (swadeley) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:51]:
```suggestion
# {foreman-maintain} content prepare
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc [hunk 13]
```diff
@@ -17,11 +17,48 @@ Cloning is the recommended method because that prevents them being overwritten i
 To confirm if a template has been edited, you can view its *History* before you upgrade or view the changes in the audit log after an upgrade.
 In the {ProjectWebUI}, navigate to *Monitor* > *Audits* and search for the template to see a record of changes made.
 If you use the export method, restore your changes by comparing the exported template and the default template, manually applying your changes.
+ifdef::satellite[]
+* Pulp content migration occurs at {ProjectVersionPrevious}, before the upgrade to {ProjectVersion}.
+* Content migration can be run online, but uses processor, disk, and memory resources.
+Sync and content view publishing operations might take longer as a result.
+* If you have not pre-migrated Pulp content, the `PULP_CONTENT_PREMIGRATION_BATCH_SIZE` setting defines the number of content units processed at the same time.
+It affects the amount of RAM used and the I/O load.
+The lower the value, the longer `satellite-maintain content-prepare` takes to complete.
```
Comment (swadeley) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc:26]:
```suggestion
The lower the value, the longer `satellite-maintain content prepare` takes to complete.
```

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+If a user attempts to halt the process using CTRL + C or by disconnecting their SSH session, the process does not terminate but continues in the background.
+You can use the following command to terminate the process gracefully, whenever necessary, so that you can continue later.
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {foreman-maintain} content-prepare abort
+----
++
+Note that `{foreman-maintain} content-prepare abort` can take several minutes to terminate the process.
+. Continue the migration process using `{foreman-maintain} content-prepare` whenever it is convenient.
```
Comment (swadeley) [inline guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:64]:
```suggestion
. Continue the migration process using `{foreman-maintain} content prepare` whenever it is convenient.
```

**Review by swadeley — CHANGES_REQUESTED** [review]
misplaced hyphens in commands

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc (hunk 9 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]: @maximiliankolb the previous discussion on this question is here: https://github.com/theforeman/foreman-documentation/pull/707#discussion_r703647893

that discussion suggests that those changes were added in error, and I have therefore removed them, the same way they were removed in #707 

those lines, if they are indeed required, can be addressed in another issue and associated commit. 

but for now, let's merge this so we can release the beta docs in a matter of days, thank you.

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc (hunk 11 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:63]: fixed

File: guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc (hunk 14 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/migrating_pulp_content.adoc:64]: fixed

File: guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc (hunk 13 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_satellite_server_prerequisites.adoc]: fixed

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc (hunk 9 above)
Comment (ryandeussing) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:156]: Just confirming that these changes are required. They are duplicative steps. Yes, they ought to be in their own PR, but Ian included them originally and I have done the same.

Duplication:
https://github.com/theforeman/foreman-documentation/blob/master/guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc

File: guides/common/assembly_provisioning-virtual-machines-kvm.adoc [hunk 15]
```diff
@@ -20,7 +20,7 @@ Ensure no other DHCP services run on this network to avoid conflicts with {Smart
 For more information about network service configuration for {SmartProxyServer}s, see {ProvisioningDocURL}Configuring_Networking[Configuring Networking].
 ifdef::satellite[]
 * A {RHEL} server running KVM virtualization tools (libvirt daemon).
-For more information, see the https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Virtualization_Getting_Started_Guide/[Red Hat Enterprise Linux 7 Virtualization Getting Started Guide].
+For more information, see the https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html-single/virtualization_getting_started_guide/[Red Hat Enterprise Linux 7 Virtualization Getting Started Guide].
```
Comment (swadeley) [inline guides/common/assembly_provisioning-virtual-machines-kvm.adoc:23]: It was, for more than ten years, customary in Red Hat not to link to `html-single` versions of guides. Has that policy changed?

Comment (ryandeussing) [reply guides/common/assembly_provisioning-virtual-machines-kvm.adoc:23]: @swadeley if that was a pattern, it is no longer in practice, there are links to html-single across Red Hat's docs. If you are familiar with any artifacts around that line of thinking, I'd be curious to understand the motivation and if it's appropriate in future we can remediate.

**Review by lzap — APPROVED** [review]
LGTM

**Comment by lzap** [comment]
CP 3.0: 8ba62b54a860884dfd109fee224d36c24622f696

CP 2.5: c57fe64b6b7cd39b4111ce2d836bb9f22c87420c
